### PR TITLE
py: properly serialize DataFrames with Timestamp columns

### PR DESCRIPTION
**Pull request by abhizer**
Fixes: #1840

Also does the following things:

* chunk dataframes into smaller groups of 1000 rows per request while ingesting data
* avoids adding empty dataframes to output buffer
* ignores the index while concatenating output dataframes

Is this a user-visible change (yes/no): no

### CODE REVIEW

File: python/feldera/_helpers.py [hunk 1]
```diff
@@ -32,3 +32,12 @@ def validate_connector_input_format(fmt: Format):
...
     if isinstance(fmt, JSONFormat) and fmt.config.get("update_format") is None:
         raise ValueError("update_format not set in the format config; consider using: .with_update_format()")
+
+
+def chunk_dataframe(df, chunk_size=1000):
+    """
+    Yield successive n-sized chunks from the given dataframe.
+    """
+
+    for i in range(0, len(df), chunk_size):
```
Comment (snkas) [inline python/feldera/_helpers.py:42]: Interesting that iloc does not throw any errors when selecting a range beyond its size.

**Review by snkas — COMMENTED** [review]
The chunking of input and the fixes are nice additions! Regarding serialization, I think it'd be useful to consider the client interface and how generic the `push_to_pipeline` function should be (or if it should be restructured / renamed / other tailored functions added). For instance, `send_request` should be kept as generic as possible, either requiring body to be bytes or having an optional serialization function passed that turns it into bytes.

File: python/feldera/output_handler.py [hunk 2]
```diff
@@ -20,7 +20,8 @@ def __init__(self, client: FelderaClient, pipeline_name: str, view_name: str, qu
 
         # the callback that is passed to the `CallbackRunner`
         def callback(df: pd.DataFrame, _: int):
-            self.buffer.append(df)
```
[the remark this answers is not in the document]
Comment (abhizer) [reply python/feldera/output_handler.py:23]: I got a warning from `pandas.concat` saying the behavior will change in the future for empty dataframes.

File: python/feldera/rest/feldera_client.py [hunk 3]
```diff
@@ -380,6 +380,7 @@ def push_to_pipeline(
             array: bool = False,
             force: bool = False,
             update_format: str = "raw",
+            dont_serialize: bool = False,
```
[the remark this answers is not in the document]
Comment (abhizer) [reply python/feldera/rest/feldera_client.py:383]: Yeah. We use JSON most of the time, maybe they should be two different functions.

**Review by ryzhyk — APPROVED** [review]
Looks good, we can work on the Feldera-compatible timestamp encoding in another PR

**Comment by ryzhyk** [comment]
@abhizer , does Pandas support Date, Time, and Decimal types?  If so, we will also need to make sure we encode those correctly.

**Comment by abhizer** [comment]
I don't think there are Date and Time separate types in Pandas. Even if it is only just the date, it seems to be DateTime and Decimals seem to be serialized as Double.

**Pull request merged**